rustls/src/server/tls12.rs

```diff
@@ -393,16 +395,16 @@ mod client_hello {
         common.send_msg(c, false);
     }
 
-    fn emit_server_kx(
+    fn emit_server_kx<C: CryptoProvider>(
         transcript: &mut HandshakeHash,
         common: &mut CommonState,
         sigschemes: Vec<SignatureScheme>,
-        skxg: &'static kx::SupportedKxGroup,
+        skxg: &'static <<C as CryptoProvider>::KeyExchange as KeyExchange>::SupportedGroup,
         signing_key: &dyn sign::SigningKey,
         randoms: &ConnectionRandoms,
-    ) -> Result<kx::KeyExchange, Error> {
-        let kx = kx::KeyExchange::start(skxg)?;
-        let secdh = ServerECDHParams::new(skxg.name, kx.pubkey.as_ref());
+    ) -> Result<C::KeyExchange, Error> {
+        let kx: <C as CryptoProvider>::KeyExchange = KeyExchange::start(skxg)?;
```

Comment:
I usually prefer to avoid type annotations everywhere. If necessary, I prefer to turbofish them (`KeyExchange::<..>::start(skxg)`). But, I think it probably shouldn't be necessary here?